libsolidity/ast/AST.cpp

```diff
@@ -363,6 +387,33 @@ FunctionDefinitionAnnotation& FunctionDefinition::annotation() const
 	return dynamic_cast<FunctionDefinitionAnnotation&>(*m_annotation);
 }
 
+FunctionDefinition::FunctionDefinition(
+	SourceLocation const &_location, ASTPointer<ASTString> const &_name,
+	Declaration::Visibility _visibility, StateMutability _stateMutability,
+	bool _isConstructor, ASTPointer<ASTString> const &_documentation,
+	ASTPointer<ParameterList> const &_parameters,
+	std::vector<ASTPointer<ModifierInvocation>> const &_modifiers,
+	ASTPointer<ParameterList> const &_returnParameters,
+	ASTPointer<Block> const &_body,
+	const ASTPointer<ModifierArea> &_modifierArea
+):
+	CallableDeclaration(_location, _name, _visibility, _parameters, _returnParameters),
+	Documented(_documentation),
+	ImplementationOptional(_body != nullptr),
+	m_stateMutability(_stateMutability),
+	m_isConstructor(_isConstructor),
+	m_body(_body),
+	m_modifierArea(_modifierArea)
+{
+	std::vector<ASTPointer<ModifierInvocation>> realModifiers = _modifiers;
+	if (m_modifierArea)
+	{
+		auto const areaModifiers = _modifierArea->totalModifiers();
+		realModifiers.insert(realModifiers.begin(), areaModifiers.begin(), areaModifiers.end());
```

Comment:
There are overloads for ``+`` and ``+=`` for vectors in libdevcore (I think) - so you should be able to just use
``realModifiers = areaModifiers + realModifiers;``.